velox/functions/lib/tests/TimeUtilsTest.cpp

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright (c) Facebook, Inc. and its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "velox/functions/lib/TimeUtils.h"
+#include <gtest/gtest.h>
+
+namespace facebook::velox::functions {
+namespace {
+
+TEST(TimeUtilsTest, fromDateTimeUnitString) {
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("", false));
+
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("microsecond", false));
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("dd", false));
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("mon", false));
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("mm", false));
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("yyyy", false));
+  ASSERT_EQ(std::nullopt, fromDateTimeUnitString("yy", false));
+
+  ASSERT_EQ(
```

Comment:
This is valid too.
```
 ASSERT_EQ(
      DateTimeUnit::kMillisecond, fromDateTimeUnitString("millisecond", false));
```
And other public functions also need tests. I'm open for this point since the function test should have cover the function you moved, but it is better to add it.